src/twisted/internet/unix.py

```diff
@@ -501,7 +501,17 @@ def _bindSocket(self):
         self.fileno = self.socket.fileno
 
     def write(self, datagram, address):
-        """Write a datagram."""
+        """
+        Write a datagram.
+
+        @param datagram: the data to be sent
+        @type datagram: L{bytes}
+
+        @param address: the address to send the datagram to
+        @type address: L{bytes}
+
+        @raise socket.error: when socket is full with arg[0] == EAGAIN
```

Comment:
FWIW: this is *definitely* the wrong fix.  `write` should never, ever raise EAGAIN.

Perhaps this could instead buffer and retry, even in the UDP case?

Reply:
At the moment its wrong in that the messages are lost.

We put the retry logic into our code and handle retry and detecting that the received is broken based on knowledge of your application requirements.

Are you saying that what is needed here is the whole producer/comsumer interface?

Reply:
Yes, that's the way Twisted deals with backpressure.

Reply:
I've added a comment to [ticket 9504](https://twistedmatrix.com/trac/ticket/9504) with some thoughts.

@barry-scott-forcepoint  I wonder if you could comment there on what your application's needs are and whether they would be well-served by a tcp-like buffer and `pauseProducing` implementation, or whether you need finer control over what to do when a write returns `EAGAIN` / `ENOBUFS` / etc.?